### PR TITLE
Fix Extern Controller Synchronization

### PR DESCRIPTION
**Pull request by DavidMansolino**
Extern controller should block the simulation while not started if the 'synchronization' field is set to `TRUE`.

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:+1: :heavy_check_mark:

**Comment by DavidMansolino** [comment]
:+1:

**Comment by omichel** [comment]
:eyes:

File: src/webots/control/WbControlledWorld.cpp [hunk 1]
```diff
@@ -340,12 +346,19 @@ void WbControlledWorld::step() {
 }
 
 bool WbControlledWorld::needToWait() {
-  foreach (WbController *const controller, mControllers)
+  if (!mRobotsWaitingExternController.isEmpty()) {
```
Comment (omichel) [inline src/webots/control/WbControlledWorld.cpp]: Is this test useful?

Comment (DavidMansolino) [reply src/webots/control/WbControlledWorld.cpp]: No not anymore :wink:

**Comment by DavidMansolino** [comment]
:+1:

**Comment by omichel** [comment]
You can merge it now 😄.

**Pull request merged**